From ce02343a1fa2815fb6184c4d08f8421bf571dedd Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Tue, 9 Aug 2016 16:50:57 -0700
Subject: [PATCH 1700/2848] YARN-5343.
 TestContinuousScheduling#testSortedNodes fails
 intermittently. (Yufei Gu via kasha)

(cherry picked from commit 7992c0b42ceb10fd3ca6c4ced4f59b8e8998e046)

Change-Id: I28d3d57b490949b9dd4c0c39d149a3340ebddf39
---
 .../scheduler/fair/TestContinuousScheduling.java   |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/TestContinuousScheduling.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/TestContinuousScheduling.java
index 1f093a5..9bca470 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/TestContinuousScheduling.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/TestContinuousScheduling.java
@@ -168,8 +168,8 @@ public void testSortedNodes() throws Exception {
     Assert.assertEquals(2, nodes.size());
   }
 
-  private void triggerSchedulingAttempt() {
-    mockClock.tickMsec(
+  private void triggerSchedulingAttempt() throws InterruptedException {
+    Thread.sleep(
         2 * scheduler.getConf().getContinuousSchedulingSleepMs());
   }
 }
-- 
1.7.9.5

